docs: getting started on android FirebaseUI v10 (#2309)
firebase/FirebaseUI-Android: GETTING_STARTED.md

    FirebaseAuthUI.getInstance().signOut(applicationContext)
}
```

```kotlin
lifecycleScope.launch {
    FirebaseAuthUI.getInstance().delete(applicationContext)
}
```

## Customization

FirebaseUI Auth is customizable, and the simplest way to get started is to set a theme directly in `authUIConfiguration`:

```kotlin
val configuration = authUIConfiguration {
    context = applicationContext
    providers {
        provider(AuthProvider.Email())
        provider(AuthProvider.Google(scopes = listOf("email"), serverClientId = null))
    }
    theme = AuthUITheme.Adaptive
}
```

You can also:

- Use `AuthUITheme.Default`, `AuthUITheme.DefaultDark`, or `AuthUITheme.Adaptive`.
- Inherit your app theme with `AuthUITheme.fromMaterialTheme()`.
- Customize the default theme with `.copy()`.
- Build a fully custom `AuthUITheme`.
- Set a logo, Terms of Service URL, and Privacy Policy URL in `authUIConfiguration`.

For full theming and customization details, including theme precedence, provider button styling, and custom themes, see [the Theming and Customization section of the readme in GitHub](https://github.com/firebase/FirebaseUI-Android/blob/master/auth/README.md#theming--customization).

> ## Existing Activity-based apps {#existing-activity-based-apps}

Reviewer: ```suggestion
## Existing Activity-based apps
```

- Material 3 UI that can inherit your app theme.
- Multiple authentication providers, including email/password, phone, Google, Facebook, Apple, GitHub, Microsoft, Yahoo, Twitter, anonymous auth, and custom OAuth.
- Multi-factor authentication support, including SMS and TOTP.
- Built-in flows for account management, account linking, and anonymous user upgrade.

## Before you begin

1. [Add Firebase to your Android project](https://firebase.google.com/docs/android/setup).
2. Make sure your app is set up for Jetpack Compose.
3. In the [Firebase console](https://console.firebase.google.com/), enable the sign-in methods you want to support.

Add FirebaseUI Auth to your app module:

```kotlin
dependencies {
    implementation("com.firebaseui:firebase-ui-auth:10.0.0-beta02")

    // Most apps also declare Firebase and Compose dependencies directly.
    implementation(platform("com.google.firebase:firebase-bom:<latest-version>"))
    implementation("com.google.firebase:firebase-auth")

    implementation(platform("androidx.compose:compose-bom:<latest-version>"))
    implementation("androidx.compose.material3:material3")

    // Required only if Facebook Login support is needed

>     implementation("com.facebook.android:facebook-login:<latest-version>")

Author: I've updated with some tweaks to the suggestion here.

> # FirebaseUI Auth for Android

Reviewer 2: Keeping the existing title

```suggestion
# Easily add sign-in to your Android app with FirebaseUI
```

> FirebaseUI Auth is a modern, Compose-based authentication library for Firebase Authentication on Android.

Reviewer 2: Let's keep the existing text?

```suggestion
[FirebaseUI](https://github.com/firebase/firebaseui-android) Auth is a library built on top of the Firebase Authentication SDK that provides drop-in UI flows for use in your app.
```

> `10.x` is currently a beta release.

Reviewer 2: ```suggestion
Caution: Version 10.x is currently a **beta release**. This means that the functionality might change in backward-incompatible ways or have limited support. A beta release is not subject to any SLA or deprecation policy.
```

Using "Caution:" will render this paragraph as a warning similar to this:

<img width="642" height="124" alt="Image" src="https://github.com/user-attachments/assets/ae396144-20c2-4b54-be21-506a2493f76e" />

> If you used the older FirebaseUI Auth guides, the biggest change in `10.x` is that the recommended sign-in flow now uses Compose screens instead of `Intent` builders and `ActivityResultLauncher` callbacks.

Reviewer 3: would it be worth adding a statement referencing this section **Existing Activity-based apps** after "builders and ActivityResultLauncher callbacks." ?

Author: Done 👍

> If you used the older FirebaseUI Auth guides, the biggest change in `10.x` is that the recommended sign-in flow now uses Compose screens instead of `Intent` builders and `ActivityResultLauncher` callbacks. For apps that still use Activities, see [Existing Activity-based apps](#existing-activity-based-apps).

Reviewer 2: ```suggestion
If you used the older FirebaseUI Auth guides, the biggest change in `10.x` is that the recommended sign-in flow now uses Compose screens instead of `Intent` builders and `ActivityResultLauncher` callbacks. For apps that still use Activities, see the [Existing Activity-based apps](#existing-activity-based-apps) section.
```

FirebaseUI Auth provides the following benefits:

- Credential Manager integration for faster sign-in on Android.
- Material 3 UI that can inherit your app theme.
- Multiple authentication providers, including email/password, phone, Google, Facebook, Apple, GitHub, Microsoft, Yahoo, Twitter, anonymous auth, and custom OAuth.
- Multi-factor authentication support, including SMS and TOTP.
- Built-in flows for account management, account linking, and anonymous user upgrade.

## Before you begin

> 1. [Add Firebase to your Android project](https://firebase.google.com/docs/android/setup).

Reviewer 2: ```suggestion
1. If you haven't already, [add Firebase to your Android project](https://firebase.google.com/docs/android/setup).
```

> 2. Make sure your app is set up for Jetpack Compose.

Reviewer 4: What does this mean? Is there a way for us to point developers to docs for them to check whether or not their app is setup for Jetpack Compose?

Reviewer 2: Let's assume developers reading this guide are already using Jetpack Compose and remove this line.

3. In the [Firebase console](https://console.firebase.google.com/), enable the sign-in methods you want to support.

> Add FirebaseUI Auth to your app module:

Reviewer 2: Should we make this another step in the list above?

```kotlin
dependencies {
    // Check Maven Central for the latest version:
    // https://central.sonatype.com/artifact/com.firebaseui/firebase-ui-auth/versions
    implementation("com.firebaseui:firebase-ui-auth:10.0.0-beta02")

    // Required: Jetpack Compose
    // Find the latest Compose BOM version here:
    // https://developer.android.com/develop/ui/compose/bom
    implementation(platform("androidx.compose:compose-bom:2026.03.00"))

>     implementation("androidx.compose.material3:material3")

Reviewer 2: I don't think we need these

    // Required only if Facebook login support is required
    // Find the latest Facebook SDK releases here: https://goo.gl/Ce5L94
    implementation("com.facebook.android:facebook-android-sdk:8.x")
}
```

> The high-level FirebaseUI Auth API is Compose-based, so if your app is not already using Compose you will need to enable it first.

Reviewer 2: We can remove this too

## Provider configuration

Some providers need additional setup before you can sign users in.

> 

Reviewer 2: What if we had a list pointing to the other docs instead of detailing the steps for each of the providers?

For example:

```suggestion
Some providers need additional setup before you can sign users in:
- [Sign in with Google](https://firebase.google.com/docs/auth/android/google-signin)
- [Facebook Login](https://firebase.google.com/docs/auth/android/facebook-login)
```

(this example is not exhaustive)

Author: done - see commit: https://github.com/firebase/FirebaseUI-Android/pull/2309/changes/bd44462e4938bc5b50c3818d748fa020a58e76a5

### Google Sign-In

> - Enable Google in the Firebase console.

Reviewer 2: ```suggestion
- Enable Google Sign-in in the Firebase console.
```

                    theme = AuthUITheme.fromMaterialTheme()
                    providers {
                        provider(AuthProvider.Email())
                        provider(
                            AuthProvider.Google(
                                scopes = listOf("email"),
                                serverClientId = null,
                            )
                        )
                    }
                }

                if (authUI.isSignedIn()) {
                    HomeScreen()
                } else {
                    FirebaseAuthScreen(
                        configuration = configuration,
                        authUI = authUI,
                        onSignInSuccess = { result ->
                            // User signed in successfully
                        },
                        onSignInFailure = { exception ->
                            // Sign in failed
                        },
                        onSignInCancelled = {
                            finish()
                        },
                    )
                }
            }
        }
    }
}
```

This gives you a complete authentication flow with:

> - Email/password sign-in and sign-up.

Reviewer 2: ```suggestion
- Password Authentication.
```
